app/src/main/java/com/example/todolistproject/view/adapter/TaskAdapter.kt

```diff
@@ -0,0 +1,78 @@
+package com.example.todolistproject.view.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.example.todolistproject.R
+import com.example.todolistproject.databinding.ItemTaskBinding
+import com.example.todolistproject.model.Task
+import java.util.*
+import javax.inject.Inject
+
+class TaskAdapter @Inject constructor() : ListAdapter<Task, TaskViewHolder>(diffUtil) {
+
+    interface OnTaskItemClickListener {
+        fun onTaskItemClick(position: Int)
+        fun onTaskItemLongClick(position: Int)
+        fun onTaskCompleteClick(position: Int)
+    }
```

Comment:
```suggestion
    interface OnTaskItemClickListener {
        fun onTaskItemClick(position: Int)
        fun onTaskItemLongClick(position: Int)
        fun onTaskCompleteClick(position: Int)
    }
```

interface 에 listener 을 하나씩 사용하면 쓰는쪽에서 람다식으로 편하게 사용할수 있습니다.
그리고 kotlin을 사용하면 HOF를 사용해서 좀더 편하게 하는것도 하나의 방법입니다.

Reply:
https://medium.com/@ccamdi/kotlin%EC%9C%BC%EB%A1%9C-%EC%9E%91%EC%84%B1%ED%95%9C-interface%EB%A5%BC-lambda%EB%A1%9C-%ED%91%9C%ED%98%84%ED%95%B4%EB%B3%B4%EC%9E%90-385425e0e954 여기 나와있는 방식을 말씀하신 건지 확인 해 주시면 감사하겠습니다!